h2o-core/src/test/java/water/runner/H2ORunner.java

```diff
@@ -28,6 +29,8 @@
 @Ignore
 public class H2ORunner extends BlockJUnit4ClassRunner {
     private final TestClass testClass;
+    private final HashSet doonlyTestsNames;
```

Comment:
The tests should be listed in fully qualified names. Therefore, we can use HashSet and do the exact match with `contains` - which is fast. 

TestUtils do `O(n*(i+d))`, where `n` is number of tests (total), `i` is the number of ignored tests listed and `d` is number of do only tests.